smserver/smserver_test.go

```diff
@@ -725,40 +736,83 @@ func TestServiceStateProcessing(t *testing.T) {
 	}
 
-	receiveNewSate, err := notifications.Recv()
+	receivedNewState, err := notifications.Recv()
 	if err != nil {
-		t.Errorf("Can't receive monitoring data: %s", err)
+		t.Fatalf("Can't receive notification data: %s", err)
+	}
+
+	if !proto.Equal(expectedNewState, receivedNewState) {
+		log.Error("received new state != sent data")
 	}
 
-	if !proto.Equal(receiveNewSate.GetNewServiceState(), etalonNewStateMsg) {
-		log.Error("received newSate data != sent data")
+	expectedStateStateAcceptance := cloudprotocol.StateAcceptance{
+		InstanceIdent: cloudprotocol.InstanceIdent{ServiceID: "service1", SubjectID: "s1", Instance: 2},
+		Checksum:      checkSumStr, Result: "result", Reason: "OK",
 	}
 
-	etalonStateRequest := &pb.ServiceStateRequest{ServiceId: "serviceId2", Default: false}
+	if _, err := smServer.InstanceStateAcceptance(context.Background(),
+		&pb.StateAcceptance{
+			Instance:      &pb.InstanceIdent{ServiceId: "service1", SubjectId: "s1", Instance: 2},
+			StateChecksum: checkSumStr, Result: "result", Reason: "OK",
+		}); err != nil {
+		t.Fatalf("Can't call instance state acceptance: %s", err)
+	}
 
-	launcher.stateChannel <- &pb.SMNotifications{
-		SMNotification: &pb.SMNotifications_ServiceStateRequest{
-			ServiceStateRequest: etalonStateRequest,
+	if !reflect.DeepEqual(expectedStateStateAcceptance, stateHandler.currentStateAcceptance) {
+		t.Errorf("Incorrect state acceptance message")
+	}
+
+	expectedStateRequest := &pb.SMNotifications{SMNotification: &pb.SMNotifications_InstanceStateRequest{
+		InstanceStateRequest: &pb.InstanceStateRequest{
+			Instance: &pb.InstanceIdent{ServiceId: "service1", SubjectId: "s1", Instance: 2}, Default: true,
 		},
+	}}
+
+	stateHandler.stateReqChan <- cloudprotocol.StateRequest{
+		InstanceIdent: cloudprotocol.InstanceIdent{ServiceID: "service1", SubjectID: "s1", Instance: 2}, Default: true,
 	}
 
-	receivedSateRequest, err := notifications.Recv()
+	receivedStateRequest, err := notifications.Recv()
 	if err != nil {
-		t.Errorf("Can't receive monitoring data: %s", err)
+		t.Fatalf("Can't receive notification data: %s", err)
+	}
+
+	if !proto.Equal(expectedStateRequest, receivedStateRequest) {
+		log.Error("received state request != sent data")
+	}
+
+	expectedUpdateState := cloudprotocol.UpdateState{
+		InstanceIdent: cloudprotocol.InstanceIdent{ServiceID: "service1", SubjectID: "s1", Instance: 2},
+		Checksum:      checkSumStr, State: stateStr,
 	}
 
-	if !proto.Equal(receivedSateRequest.GetServiceStateRequest(), etalonStateRequest) {
-		log.Error("received state request data != sent data")
+	if _, err := smServer.SetInstanceState(context.Background(), &pb.InstanceState{
+		Instance:      &pb.InstanceIdent{ServiceId: "service1", SubjectId: "s1", Instance: 2},
+		StateChecksum: checkSumStr,
+		State:         []byte(stateStr),
+	}); err != nil {
+		t.Fatalf("Can't set instance state: %s", err)
 	}
+
+	reflect.DeepEqual(expectedUpdateState, stateHandler.currentUpdateState)
```

Comment:
should be checked? and I can't find where currentUpdateState is set

Reply:
fixed